ntbs-integration-tests/NotificationPages/ClinicalDetailsPageTests.cs

```diff
@@ -97,6 +97,30 @@ public async Task PostDraft_ReturnsPageWithAllModelErrors_IfModelNotValid()
             resultDocument.AssertErrorSummaryMessage("ClinicalDetails-Notes", "notes", "Too many characters entered into the Notes field");
         }
 
+        [Fact]
+        public async Task PostDraft_ReturnsPageWithTabError_IfNotesContainTabs()
+        {
+            // Arrange
+            var url = GetCurrentPathForId(Utilities.DRAFT_ID);
+            var initialDocument = await GetDocumentForUrlAsync(url);
+
+            var formData = new Dictionary<string, string>
+            {
+                ["NotificationId"] = Utilities.DRAFT_ID.ToString(),
+                ["NotificationSiteMap[PULMONARY]"] = "true",
+                ["ClinicalDetails.Notes"] = "Notes\t",
+            };
+
+            // Act
+            var result = await Client.SendPostFormWithData(initialDocument, formData, url);
+
+            // Assert
+            var resultDocument = await GetDocumentAsync(result);
+
+            result.EnsureSuccessStatusCode();
+            resultDocument.AssertErrorSummaryMessage("ClinicalDetails-Notes", "notes", "Notes cannot contain tabs. Use spaces instead");
+        }
+

```

Comment:
I've added the `ContainsNoTabs` attribute to a bunch of models, but only added integration tests here and for `EditContactDetailsTests`. I did this because to test this we need to add another integration test for each page, because of the way the validation works they only test one validation rule per field, per test, and we don't really have full coverage of all of our validation attributes anyway. If you want me to add them for the rest of the classes I definitely can do so. What do you thing?

Reply:
Yes, I think having an integration test for every possible error is overkill. With tests you have to tradeoff how long they take to run against the amount of coverage they give. In this case those extra tests really wouldn't add much extra coverage (we already have other tests which verify that the correct attribute is applied to those fields), so I don't think they're worth it.